webapp-django/crashstats/crashstats/models.py

```diff
@@ -1368,6 +1377,51 @@ class SignatureFirstDate(SocorroMiddleware):
         )
     }
 
+    def get_dates(self, signatures):
+        """A highly optimized version, that returns a dictionary where
+        the keys are the signature and the values are dicts that look
+        like this for example::
+
+            {
+                'first_build': u'20101214170338',
+                'first_date': datetime.datetime(
+                    2011, 1, 17, 21, 24, 18, 979850, tzinfo=...)
+                )
+            }
+
+        """
+        dates = {}
+        missing = {}
+        for signature in signatures:
+            # calculate a good cache key for each signature
+            cache_key = 'signature_first_date-{}'.format(
+                hashlib.md5(signature.encode('utf-8')).hexdigest()
+            )
+            cached = cache.get(cache_key)
+            if cached is not None:
+                dates[signature] = cached
+            else:
+                # remember the cache keys of those we need to look up
+                missing[signature] = cache_key
+
+        if missing:
+            hits = super(SignatureFirstDate, self).get(
+                signatures=missing.keys()
+            )['hits']
+
+            for hit in hits:
+                signature = hit.pop('signature')
```

Comment:
What's the point of using ``pop`` here? Is there any performance gain compared to ``get``? I can't see any difference locally, and ``pop`` is more confusing in my opinion.